Source/FullScreenPokemon.ts

```diff
@@ -1989,6 +1989,19 @@ module FullScreenPokemon {
                 thing.sightDetector.nocollide = true;
             }
 
+            if (thing.surfing && thing.bordering[thing.direction] && thing.bordering[thing.direction].title.indexOf("WaterEdge") !== -1) {
```

Comment:
Yeah, there's a better way to do this.

Give WaterEdge a .collide function that returns true if it's colliding with a surfing player.